lib/adhearsion/call_controller/dial.rb

```diff
@@ -191,7 +193,7 @@ def prep_calls
                 condition.wait
               end
 
-              if new_call.alive? && new_call.active? && status.result != :answer
+              if new_call.alive? && new_call.active? && status.result != :answer && @call.alive? && @call.active?
                 logger.info "#dial joining call #{new_call.id} to #{@call.id}"
                 pre_join_tasks new_call
```

Comment:
Thank you for doing this.  But my bad... I'm sorry that I keep on changing my mind...  My gut says that we need to run `pre_join_tasks new_call` even if @call is dead.  What do you think about moving things around like this?

```ruby
if new_call.alive? && new_call.active? && status.result != :answer
  pre_join_tasks new_call

  if @call.alive? && @call.active?
    logger.info "#dial joining call #{new_call.id} to #{@call.id}"  
    @call.answer
    ...
```

Reply:
sounds good to me - pushed the change back out, @sfgeorge